fiona/collection.py

```diff
@@ -50,7 +52,8 @@ def __init__(self, path, mode='r', driver=None, schema=None, crs=None,
             raise TypeError("invalid driver: %r" % driver)
         if schema and not hasattr(schema, 'get'):
             raise TypeError("invalid schema: %r" % schema)
-        if crs and not isinstance(crs, (dict,) + string_types):
+        if crs and not isinstance(
+                crs, (dict, collections.UserDict) + string_types):
```

Comment:
I'm not a fan of `isinstance` checks for specific subclasses; makes code brittle. What if a user a user constructs the crs as an `OrderedDict` or some other mapping? Unless there is a specific reason not to support them, all mappings could be valid here. 

How about testing against the parent class - all dictionaries should inherit from `collections.Mapping`:

```
import collections
assert isinstance({},collections.Mapping)
assert isinstance(collections.UserDict({}), collections.Mapping)
assert isinstance(collections.OrderedDict({}), collections.Mapping)
```